File: Java/calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        char op;
        double num1, num2;

        System.out.print("Ingrese el primer número: ");
        num1 = scanner.nextDouble();
        System.out.print("Ingrese el operador (+, -, *, /): ");
        op = scanner.next().charAt(0);
        System.out.print("Ingrese el segundo número: ");
        num2 = scanner.nextDouble();

        switch (op) {
            case '+':
                System.out.println("Resultado: " + (num1 + num2));
                break;
            case '-':
                System.out.println("Resultado: " + (num1 - num2));
                break;
            case '*':
                System.out.println("Resultado: " + (num1 * num2));
                break;
            case '/':
                if (num2 != 0) {
                    System.out.println("Resultado: " + (num1 / num2));
                } else {
                    System.out.println("No es posible dividir por cero.");
                }
                break;
            default:
                System.out.println("Operador no válido.");
        }

        scanner.close();
    }
}

File: Swift/Calculator.swift
import Foundation

print("Ingrese el primer número:")
guard let num1Input = readLine(), let num1 = Double(num1Input) else {
    fatalError("Entrada inválida. Por favor, ingrese un número.")
}

print("Ingrese el operador (+, -, *, /):")
guard let op = readLine() else {
    fatalError("Entrada inválida. Por favor, ingrese un operador.")
}

print("Ingrese el segundo número:")
guard let num2Input = readLine(), let num2 = Double(num2Input) else {
    fatalError("Entrada inválida. Por favor, ingrese un número.")
}

switch op {
case "+":
    print("Resultado: \(num1 + num2)")
case "-":
    print("Resultado: \(num1 - num2)")
case "*":
    print("Resultado: \(num1 * num2)")
case "/":
    if num2 != 0.0 {
        print("Resultado: \(num1 / num2)")
    } else {
        print("No es posible dividir por cero.")
    }
default:
    print("Operador no válido.")
}
